features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/internal/FlagsStateManager.kt

```diff
@@ -33,19 +35,26 @@ internal class FlagsStateManager(
     private val executorService: ExecutorService,
     private val internalLogger: InternalLogger
 ) : StateObservable {
+    /**
+     * Fair lock to ensure FIFO ordering of state mutations and listener operations.
+     * The fair parameter ensures that threads acquire the lock in the order they requested it.
+     */
```

Comment:
> threads acquire the lock in the order they requested it

Does it really matter actually? I'm wondering if it is better to use `ReadWriteLock` since probably the number of read operations will dominate over the number of write operations, or is it a wrong assumption?